docs/API.md

```diff
@@ -100,6 +100,16 @@ echo $SPACESHIP_ROOT
 #> /path/to/spaceship-prompt
 ```
 
+## `SPACESHIP_CUSTOM_SECTION_LOCATION`
+
+Path where custom section files are auto sourced. Defaults to `$HOME/.config/spaceship/sections`.
+
+Custom section name and function name should follow the same pattern as the built/core sections. To be specific, a custom section named `foobar` should be rendered with function `spaceship_foobar` and stored in file `${SPACESHIP_CUSTOM_SECTION_LOCATION}/foobar.zsh`.
```

Comment:
(not a maintainer so take with a grain of salt) Saw this PR after proposing almost the same thing as this part in #763. Two questions:

- Since the path is an option, would it make sense to put it in Options.md?
- Did you consider using a variable that points to the parent of the `sections` directory, i.e default `$HOME/.config/spaceship`, and hardcoding the `/sections`? It's a little less flexible for this specific instance, but it would allow future Spaceship enhancements to use other subdirectories like `~/.config/spaceship/some-future-need` without having to set new variables for each one.